File: backend/src/main/java/es/dawgrupo2/zendashop/extendedDTO/UserExtendedDTO.java
package es.dawgrupo2.zendashop.extendedDTO;

import java.time.LocalDateTime;
import java.util.List;

import es.dawgrupo2.zendashop.basicDTO.ImageDTO;
import es.dawgrupo2.zendashop.basicDTO.OpinionBasicDTO;
import es.dawgrupo2.zendashop.basicDTO.OrderBasicDTO;

public record UserExtendedDTO(
    Long id,
    String name,
    String surname,
    String email,
    String address,
    LocalDateTime creationDate,
    List<String> roles,
    OrderBasicDTO cart,
    ImageDTO avatar,
    List<OpinionBasicDTO> opinions,
    List<OrderBasicDTO> orders
) {
    
}

File: frontend/app/dtos/UserExtendedDTO.ts
import type ImageDTO from "./ImageDTO";
import type OrderBasicDTO from "./OrderBasicDTO";

export default interface UserExtendedDTO {
  id: number;
  name: string;
  surname: string;
  email: string;
  address: string;
  creationDate: string; 
  encodedPassword: string;
  roles: string[];
  cart: OrderBasicDTO;
  avatar: ImageDTO;
}
